lib/src/mixins/request_handling.dart

```diff
@@ -1,23 +1,95 @@
+import 'dart:convert';
+
+import 'package:dio/dio.dart';
+import 'package:http_mock_adapter/http_mock_adapter.dart';
 import 'package:http_mock_adapter/src/exceptions.dart';
 import 'package:http_mock_adapter/src/handlers/request_handler.dart';
+import 'package:http_mock_adapter/src/mixins/mixins.dart';
 import 'package:http_mock_adapter/src/request.dart';
 import 'package:http_mock_adapter/src/response.dart';
 import 'package:http_mock_adapter/src/types.dart';
 
 /// [RequestHandling] exposes developer-friendly methods which take in route,
 /// [Request], both of which ultimately get processed by [RequestHandler].
-mixin RequestHandling {
+mixin RequestHandling on Recording {
+  Dio get dio;
+
+  /// Configures default headers which are usually set by [DioMixin].
+  /// * content-type
+  /// * content-length
+  Future<void> setDefaultRequestHeaders(Dio dio, RequestOptions options) async {
+    final data = options.data;
+    if (data != null &&
+        RequestMethods.allowedPayloadMethods
+            .contains(RequestMethods.forName(name: options.method))) {
+      if (data is FormData) {
+        options.headers[Headers.contentTypeHeader] =
+            'multipart/form-data; boundary=${data.boundary}';
+        options.headers[Headers.contentLengthHeader] = data.length.toString();
+      } else {
+        final _data = await dio.transformer.transformRequest(options);
+        List<int> bytes;
+        if (options.requestEncoder != null) {
+          bytes = options.requestEncoder!(_data, options);
+        } else {
+          bytes = utf8.encode(_data);
+        }
+        options.headers[Headers.contentLengthHeader] = bytes.length.toString();
+        options.headers.putIfAbsent(
+          Headers.contentTypeHeader,
+          () => Headers.jsonContentType,
```

Comment:
I think the json content header is not set in `DioMixin` but in `Options.compose` so it is in all cases already configured here. This can probably be removed.